repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -464,6 +464,18 @@ public EntityMutationResponse mapAttributesAndClassifications(EntityMutationCont
                         }
 
                         addHasLineage(inOutEdges, isRestoreEntity);
+                    } else {
+                        if (CollectionUtils.isNotEmpty(context.getEntitiesToRestore()) && context.getEntitiesToRestore().contains(vertex)) {
+                            Set<AtlasEdge> activeEdges = new HashSet<>();
+                            Iterator<AtlasEdge> iterator = vertex.getEdges(AtlasEdgeDirection.BOTH).iterator();
+                            while (iterator.hasNext()) {
+                                AtlasEdge edge = iterator.next();
+                                if (edge.getProperty(Constants.STATE_PROPERTY_KEY, String.class).equalsIgnoreCase("ACTIVE")) {
+                                    activeEdges.add(edge);
+                                }
+                            }
+                            addHasLineage(activeEdges, true);
```

Comment:
call addHasLineage method only when activeEdges size is more than 1, make sense to add size check

Reply:
definitely

Reply:
addressed